Added names to sequence adaptations

cohansen opened the pull request:
Adds unique sequence adaptation names to conform to the standard of our other dictionary types.

To test:
When uploading a `.js` file on the Dictionaries page a new name field appears where a required name is needed for Sequence Adaptations. This was done because the adaptations are built files and don't contain metadata like the other dictionaries.

Backend PR: https://github.com/NASA-AMMOS/aerie/pull/1469

cohansen committed:
Added names to sequence adaptations

cohansen committed:
Removed unnecessary cancel button

cohansen committed:
Removed the skip on sequence adaptation creation test

cohansen committed:
Fixed some issues with the seq name tests

AaronPlave commented on src/routes/dictionaries/+page.svelte:
```diff
@@ -30,16 +30,31 @@
+      file = files[0];
 
-  async function uploadDictionaryOrAdaptation(files: FileList) {
+      isSequenceAdaptation = file.type === 'application/x-javascript';
```
I'm trying to upload an `adaptation.js` file but it looks like the type is `text/javascript` instead of `application/x-javascript`. Any reason not to just check for a `.js` extension instead?

cohansen replied:
The check for `file.type` was just cleaner, I can change it to check against the extension instead.

AaronPlave commented on src/routes/dictionaries/+page.svelte:
```diff
@@ -30,16 +30,31 @@
+  let sequenceAdaptationName: string;
 
   $: hasCreatePermission = featurePermissions.commandDictionary.canCreate(data.user);
   $: createButtonDisabled = !files;
```
Can we disable create button when `isSequenceAdaptation` and there is no sequence adaptation name?

AaronPlave commented on src/routes/dictionaries/+page.svelte:
Could we clear the form after successful upload?

cohansen committed:
PR fixes

Changed file type check to be for .js, disabled create button when a name isn't provided, clear the input form on creation

AaronPlave commented on src/routes/dictionaries/+page.svelte:
```diff
@@ -30,16 +30,32 @@
-  $: createButtonDisabled = !files;
+  $: createButtonDisabled = !files || (isSequenceAdaptation && sequenceAdaptationName === '');
+  $: {
+    if (files) {
```
Recommend `files && files.length > 0` since `files` is an object regardless of number of files. If the user hits escape when selecting a file this will cause an error here since `file` will be undefined.

AaronPlave commented on src/routes/dictionaries/+page.svelte:
```diff
@@ -30,16 +30,32 @@
 
   $: hasCreatePermission = featurePermissions.commandDictionary.canCreate(data.user);
-  $: createButtonDisabled = !files;
+  $: createButtonDisabled = !files || (isSequenceAdaptation && sequenceAdaptationName === '');
```
Need to check for existence of a file within `files` instead of just the existence of the potentially empty `files` object. Otherwise, the create button will not be properly disabled in all relevant cases.

cohansen committed:
Added a check for files length for create button disabling and seq adaptation checking

AaronPlave reviewed (approved):
LGTM!
